addons/website_sale/models/product_product.py

```diff
@@ -88,7 +88,9 @@ def _is_sold_out(self):
         return combination_info['product_type'] == 'product' and combination_info['free_qty'] <= 0
 
     def _website_show_quick_add(self):
-        return self.sale_ok
+        website = self.env['website'].get_current_website()
+        prevent_zero_price_sale = website and website.prevent_zero_price_sale
+        return self.sale_ok and (self._get_contextual_price() or not prevent_zero_price_sale)
```

Comment:
FYI `get_current_website` will always return a record, empty or not.
Also FYI accessing a field on an empty recordset will yield a falsy value, as such your condition here is not really necessary.
```suggestion
        website = self.env['website'].get_current_website()
        return self.sale_ok and (self._get_contextual_price() or not website.prevent_zero_price_sale)
```